src/editor/components/components/Sidebar.js

```diff
@@ -63,86 +72,126 @@ export default class Sidebar extends React.Component {
     this.setState({ showSideBar: !this.state.showSideBar });
   };
 
+  renderSidebarContent() {
+    const { entity } = this.props;
+
+    if (entity.id === 'reference-layers') {
+      return <GeoSidebar entity={entity} />;
+    }
+
+    if (entity.getAttribute('street-segment')) {
+      return (
+        <>
+          <StreetSegmentSidebar entity={entity} />
+          <div className="advancedComponentsContainer">
+            <AdvancedComponents entity={entity} />
+          </div>
+        </>
+      );
+    }
+
+    if (entity.getAttribute('managed-street')) {
+      return (
+        <>
+          <ManagedStreetSidebar entity={entity} />
+          <div className="advancedComponentsContainer">
+            <AdvancedComponents entity={entity} />
+          </div>
+        </>
+      );
+    }
+
+    const isIntersection = entity.getAttribute('intersection');
+    const hasNoTransform = entity.hasAttribute('data-no-transform');
+
+    return (
+      <>
+        {entity.mixinEls.length > 0 && <Mixins entity={entity} />}
+
+        {!hasNoTransform && (
+          <div id="sidebar-buttons">
+            <Button variant={'toolbtn'} onClick={() => cloneEntity(entity)}>
+              Duplicate
+            </Button>
+            <Button variant={'toolbtn'} onClick={() => removeSelectedEntity()}>
+              Delete
+            </Button>
+          </div>
+        )}
+
+        {isIntersection && <IntersectionSidebar entity={entity} />}
+        <ComponentsContainer entity={entity} />
+      </>
+    );
+  }
+
+  renderCollapsedSidebar(entity, displayName) {
```

Comment:
nit:  id prefer either having this in the component render or as a separate component

Reply:
agree but will leave this for a future task to optimize sidebar